<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>5</x>
      <y>5</y>
      <width>1271</width>
      <height>696</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>16777215</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>battery</string>
     </attribute>
     <widget class="QTableWidget" name="tableWidget">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>45</y>
        <width>301</width>
        <height>236</height>
       </rect>
      </property>
      <row>
       <property name="text">
        <string>status</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>voltage</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>current</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>temperature</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>level</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>capacity</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>err code</string>
       </property>
      </row>
      <column>
       <property name="text">
        <string>battery-1</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>battery-2</string>
       </property>
      </column>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>455</y>
        <width>201</width>
        <height>81</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox{
border-width:2px;
border-style:solid;
border-radius: 10px;
border-color:gray;
margin-top:1ex;
}
QGroupBox::title{
subcontrol-origin:margin;
subcontrol-position:top left;
left:10px;
margin-left:0px;
padding:0 1px;
}</string>
      </property>
      <property name="title">
       <string>battery-2</string>
      </property>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>175</x>
         <y>60</y>
         <width>21</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>0A</string>
       </property>
      </widget>
      <widget class="QSlider" name="setcurrent_slider_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>160</width>
         <height>16</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkBox_bat2_enable">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>30</y>
         <width>71</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>enable</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkBox_bat2_charge">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>71</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>charge</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>345</y>
        <width>201</width>
        <height>81</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox{
border-width:2px;
border-style:solid;
border-radius: 10px;
border-color:gray;
margin-top:1ex;
}
QGroupBox::title{
subcontrol-origin:margin;
subcontrol-position:top left;
left:10px;
margin-left:0px;
padding:0 1px;
}</string>
      </property>
      <property name="title">
       <string>battery-1</string>
      </property>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>175</x>
         <y>60</y>
         <width>21</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>0A</string>
       </property>
      </widget>
      <widget class="QSlider" name="setcurrent_slider_1">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>160</width>
         <height>16</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkBox_bat1_enable">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>30</y>
         <width>71</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>enable</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkBox_bat1_charge">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>71</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>charge</string>
       </property>
      </widget>
     </widget>
     <widget class="RealTimeCurve" name="rtcurve_bat_v" native="true">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>40</y>
        <width>856</width>
        <height>311</height>
       </rect>
      </property>
     </widget>
     <widget class="RealTimeCurve" name="rtcurve_bat_t" native="true">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>360</y>
        <width>856</width>
        <height>286</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>heater</string>
     </attribute>
     <widget class="QTableWidget" name="tableWidget_2">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>35</y>
        <width>221</width>
        <height>176</height>
       </rect>
      </property>
      <row>
       <property name="text">
        <string>status</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>temperature</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>pt100 adc_code</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>set point</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>duty cycle</string>
       </property>
      </row>
      <column>
       <property name="text">
        <string>heater1&amp;2</string>
       </property>
      </column>
     </widget>
     <widget class="QWidget" name="rtcurve_heater_t" native="true">
      <property name="geometry">
       <rect>
        <x>265</x>
        <y>15</y>
        <width>996</width>
        <height>631</height>
       </rect>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>325</y>
        <width>191</width>
        <height>231</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox{
border-width:2px;
border-style:solid;
border-radius: 10px;
border-color:gray;
margin-top:1ex;
}
QGroupBox::title{
subcontrol-origin:margin;
subcontrol-position:top left;
left:10px;
margin-left:0px;
padding:0 1px;
}</string>
      </property>
      <property name="title">
       <string>settings</string>
      </property>
      <widget class="QPushButton" name="pushButton_3">
       <property name="geometry">
        <rect>
         <x>95</x>
         <y>195</y>
         <width>89</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>download</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_2">
       <property name="geometry">
        <rect>
         <x>85</x>
         <y>85</y>
         <width>71</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Kp</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_3">
       <property name="geometry">
        <rect>
         <x>85</x>
         <y>120</y>
         <width>71</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Ki</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_4">
       <property name="geometry">
        <rect>
         <x>85</x>
         <y>155</y>
         <width>71</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Kd</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>55</x>
         <y>90</y>
         <width>26</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Kp</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>55</x>
         <y>125</y>
         <width>26</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Ki</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>55</x>
         <y>160</y>
         <width>26</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Kd</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_5">
       <property name="geometry">
        <rect>
         <x>85</x>
         <y>50</y>
         <width>71</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>set point</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>15</x>
         <y>55</y>
         <width>66</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>set point</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton">
       <property name="geometry">
        <rect>
         <x>15</x>
         <y>25</y>
         <width>41</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>on</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_2">
       <property name="geometry">
        <rect>
         <x>65</x>
         <y>25</y>
         <width>46</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>off</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_6">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>195</y>
         <width>66</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>upload</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>fw update</string>
     </attribute>
     <widget class="QPushButton" name="pushButton">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>70</y>
        <width>96</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>browse file</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit">
      <property name="geometry">
       <rect>
        <x>35</x>
        <y>70</y>
        <width>191</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_2">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>110</y>
        <width>96</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>update start</string>
      </property>
     </widget>
     <widget class="QProgressBar" name="progressBar">
      <property name="geometry">
       <rect>
        <x>35</x>
        <y>160</y>
        <width>291</width>
        <height>23</height>
       </rect>
      </property>
      <property name="value">
       <number>24</number>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_4">
    <property name="geometry">
     <rect>
      <x>1200</x>
      <y>705</y>
      <width>61</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>exit</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_5">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>705</y>
      <width>61</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>reset</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>RealTimeCurve</class>
   <extends>QWidget</extends>
   <header location="global">realtimecurve.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
